### PR TITLE
Sonic search support

### LINKED ISSUE #5096: Have a replacement for elasticsearch.
## Summary

Elasticsearch is really resource intensive and is kinda slow.
Having support for something like https://github.com/valeriansaliou/sonic which I use in my other projects and am really happy with it's performance.
It'd make it easier for people with more underpowered computers.

### PR DESCRIPTION
**Pull request by ThatOneCalculator**
Continuation of #5138

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +39,8 @@
 		"gulp-rename": "2.0.0",
 		"gulp-replace": "1.1.3",
 		"gulp-terser": "2.1.0",
-		"js-yaml": "4.1.0"
+		"js-yaml": "4.1.0",
+		"sonic-channel": "1.2.6"
```
Comment (syuilo) [inline package.json]: It should be in backend/package.json.

File: packages/backend/src/server/api/endpoints/notes/search.ts [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-import define from '../../define';
-import { Notes } from '@/models/index';
-import { In } from 'typeorm';
-import { ID } from '@/misc/cafy-id';
-import config from '@/config/index';
-import { makePaginationQuery } from '../../common/make-pagination-query';
-import { generateVisibilityQuery } from '../../common/generate-visibility-query';
-import { generateMutedUserQuery } from '../../common/generate-muted-user-query';
-import { generateBlockedUserQuery } from '../../common/generate-block-query';
+import $ from "cafy";
```
Comment (syuilo) [inline packages/backend/src/server/api/endpoints/notes/search.ts]: use `'`

**Comment by ThatOneCalculator** [comment]
@syuilo fixed :)

File: packages/backend/src/server/api/endpoints/notes/search.ts [hunk 3]
```diff
@@ -50,37 +50,44 @@ export const meta = {
...
+			type: "object",
+			optional: false,
+			nullable: false,
+			ref: "Note",
 		},
 	},
 
-	errors: {
-	},
+	errors: {},
```
Comment (syuilo) [inline packages/backend/src/server/api/endpoints/notes/search.ts:28]: Please do not include such unnecessary changes as they will interfere with the review. 🙏

File: packages/backend/src/server/api/endpoints/notes/search.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import $ from 'cafy';
-import es from '../../../../db/elasticsearch';
+import searchClient from '../../../../db/searchClient';
```
Comment (Johann150) [inline packages/backend/src/server/api/endpoints/notes/search.ts]: Use `@/` for the root directory of the backend package like this:
```suggestion
import searchClient from '@/db/searchClient';
```

File: packages/backend/src/server/api/endpoints/notes/search.ts (hunk 3 above)
Comment (Johann150) [reply packages/backend/src/server/api/endpoints/notes/search.ts:64]: The same goes for `res` and other changes in this file where single quotes were replaced with double quotes.

File: packages/backend/src/services/note/create.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import * as mfm from 'mfm-js';
-import es from '../../db/elasticsearch';
+import searchClient from "../../db/searchClient";
```
Comment (Johann150) [inline packages/backend/src/services/note/create.ts]:
```suggestion
import searchClient from '@/db/searchClient';
```

File: packages/backend/src/server/api/endpoints/notes/search.ts (hunk 3 above)
Comment (ThatOneCalculator) [reply packages/backend/src/server/api/endpoints/notes/search.ts:28]: Sorry about that, the code formatter did that.

**Comment by ThatOneCalculator** [comment]
@syuilo @Johann150 updated

**Comment by syuilo** [comment]
The code style is not consistent with this repository, so it needs to be rewritten a bit. I will do it, but it will take some time.

**Comment by ThatOneCalculator** [comment]
Thank you so much, it means a lot :heart:

**Comment by ThatOneCalculator** [comment]
Currently stuck on 
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/misskey/misskey/packages/backend/built/db/searchClient' imported from /home/misskey/misskey/packages/backend/built/server/api/endpoints/notes/search.js
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:418:11)
    at moduleResolve (node:internal/modules/esm/resolve:981:10)
    at defaultResolve (node:internal/modules/esm/resolve:1078:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)
    at link (node:internal/modules/esm/module_job:78:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

trying to fix.

**Comment by ThatOneCalculator** [comment]
The file that really needs work done is `packages/backend/src/server/api/endpoints/notes/search.ts`.
